### PR TITLE
Get rid of legacy class StreamWriter #2109

### PR DESCRIPTION
**Pull request by pfreixes**
## What do these changes do?

Legacy StreamWriter class as a pure proxy of the transport and the protocol is
no longer needed. All of the functionalities that were behind this class
has been moved to the `PayloadWriter` class.

Some changes that have to be considered that impacted by this change

* TCP Operations have been isolated in a module rather than move them
into the `PayloadWriter` class. All of these operations have been simplified and state less.

* `WebSocketWriter` class had a dependency with the StreamWriter, to get rid of
that dependency the constructor has been modified to take the protocol
and the transport.

A next step changing the name of the `PayLoadWriter` class for the `StreamWriter` - it reborn!- might be considered, making it consistent with the reader side.

## Are there changes in behavior for the user?

No

## Related issue number

#2109

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by fafhrd91** [comment]
looks good to me

File: aiohttp/http_writer.py [hunk 1]
```diff
@@ -110,11 +30,15 @@ def __init__(self, stream, loop):
...
-        self._transport = self._stream.transport
 
-    async def get_transport(self):
+    @property
+    def transport(self):
         return self._transport
 
+    @property
+    def protocol(self):
+        return self._protocol
```
Comment (asvetlov) [inline aiohttp/http_writer.py:40]: Please add a test for the property

Comment (pfreixes) [reply aiohttp/http_writer.py:40]: Done.

**Comment by asvetlov** [comment]
Another question: why do you return a value from `tcp_cork`/`tcp_nodelay` but never use it in aiohttp code?
Let's drop this functionality.
Tests can check a real TCP option by `getsockopt` call instead of checking the return value.

**Comment by pfreixes** [comment]
@asvetlov removed the value returned by the TCP operations. My fault Ive ended up changing the code to have an easy way to test the functions, once these have become stateless there is some edge cases where I cant see how to assert properly. For example the case of the socket `None`. 
Any idea how to improve this [1]?

[1] https://github.com/aio-libs/aiohttp/pull/2651/commits/87928bea98956da46663efcd394cb5e33e245b8d#diff-ef69e535e777102409aee677f44b4137R75

**Comment by asvetlov** [comment]
You can monkey-patch socket object to make sure that its methods are not called but I think current code is good enough.

**Comment by asvetlov** [comment]
Thanks!
